go/s3/UploadObject/UploadObject_test.go

```diff
@@ -0,0 +1,158 @@
+
+   This file is licensed under the Apache License, Version 2.0 (the "License").
+   You may not use this file except in compliance with the License. A copy of
+   the License is located at
+
+    http://aws.amazon.com/apache2.0/
+
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+package main
+
+import (
+    "encoding/json"
+    "io/ioutil"
+    "testing"
+    "time"
+
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/service/s3"
+    "github.com/aws/aws-sdk-go/service/s3/s3manager"
+
+    "github.com/google/uuid"
+)
+
+type Config struct {
+    Bucket   string `json:"Bucket"`
+    Filename string `json:"Filename"`
+}
+
+var configFileName = "config.json"
+
+var globalConfig Config
+
+func populateConfiguration(t *testing.T) error {
+    content, err := ioutil.ReadFile(configFileName)
+    if err != nil {
+        return err
+    }
+
+    text := string(content)
+
+    err = json.Unmarshal([]byte(text), &globalConfig)
+    if err != nil {
+        return err
+    }
+
+    t.Log("Bucket:    " + globalConfig.Bucket)
+    t.Log("Filename: " + globalConfig.Filename)
+
+    return nil
+}
+
+func createBucket(sess *session.Session, bucket *string) error {
+    svc := s3.New(sess)
+
+    _, err := svc.CreateBucket(&s3.CreateBucketInput{
+        Bucket: bucket,
+    })
+    if err != nil {
+        return err
+    }
+
+    err = svc.WaitUntilBucketExists(&s3.HeadBucketInput{
+        Bucket: bucket,
+    })
+    if err != nil {
+        return err
+    }
+
+    return nil
+}
+
+func deleteBucket(sess *session.Session, bucket *string) error {
+    svc := s3.New(sess)
+
+    iter := s3manager.NewDeleteListIterator(svc, &s3.ListObjectsInput{
+        Bucket: bucket,
+    })
+
+    err := s3manager.NewBatchDeleteWithClient(svc).Delete(aws.BackgroundContext(), iter)
+    if err != nil {
+        return err
+    }
+
+    _, err = svc.DeleteBucket(&s3.DeleteBucketInput{
+        Bucket: bucket,
+    })
+    if err != nil {
+        return err
+    }
+
+    err = svc.WaitUntilBucketNotExists(&s3.HeadBucketInput{
+        Bucket: bucket,
+    })
+    if err != nil {
+        return err
+    }
+
+    return nil
+}
+
+func TestUploadObject(t *testing.T) {
+    thisTime := time.Now()
+    nowString := thisTime.Format("2006-01-02 15:04:05 Monday")
+    t.Log("Starting unit test at " + nowString)
+
+    err := populateConfiguration(t)
+    if err != nil {
+        t.Fatal(err)
+    }
+
+    sess := session.Must(session.NewSessionWithOptions(session.Options{
+        SharedConfigState: session.SharedConfigEnable,
+    }))
+
+    createdBucket := false
+
+    if globalConfig.Bucket == "" {
+        // Create a unique, random name (import "github.com/google/uuid"):
```

Comment:
is colon needed to end comment? if not delete

Reply:
I've deleted the comment.